README.md

```diff
@@ -107,6 +107,54 @@ Notes:
 ## Assets
 The Copenhagen theme doesn't have any assets, but you can add assets to your theme by placing them in the `assets` folder.
 
+# React components
+
+## Internationalization
+
+I18n is implemented in the React components using the [react-i18next](https://react.i18next.com/) library. We use a flat JSON file and we use `.` as a separator for plurals, which is different from the default `_` and it is configured during initialization.
+
+We also added some tools to be able to integrate the library with the internal translation system used at Zendesk. If you are building a custom theme and you want to provide your own translations you can refer to the library documentation to setup the loading of your translations.
+
+### Integration with the Zendesk translation system
+
+#### Adding translations strings
+
+Translation strings are added directly in the source code, usually using the `useTranslation` hook, passing the key and the default English value:
+
+```ts
+import { useTranslation } from 'react-i18next';
+
+function MyComponent() {
+  const { t } = useTranslation();
+
+  return <div>{t("my-key", "My default value")}</div>
+}
+```
+
+Providing the default English value in the code makes it possible to use it as a fallback value when strings are not yet translated and to extract the strings from the source code to the translations YAML file.
+
+#### Plurals 
+When using [plurals](https://www.i18next.com/translation-function/plurals), we need to provide default values for the `zero`, `one`, and `other` values, as requested by our translation system. This can be done by passing the default values in the [options](https://www.i18next.com/translation-function/essentials#overview-options) of the `t` function.
+
+```ts
+t("my-key", {
+  "defaultValue.zero": "{{count}} items",
+  "defaultValue.one": "{{count}} item",
+  "defaultValue.other": "{{count}} items",
+  count: ...
+})
+```
+
+#### String extraction
+
+The `bin/extract-strings.mjs` script can be used to extract translation strings from the source code and put them in the YAML file that is picked up by our internal translation system. The usage of the script is documented in the script itself.
+
+The script wraps the `i18next-parser` tool and converts its output to the YAML format used internally. It is possible to use a similar approach in a custom theme, either using the standard `i18next-parser` output as the source for translations or implementing a custom transformer.
+
+#### Updating translation files
+
+Use the `bin/update-module-translations.mjs` to download the latest translations for a module. All files are then bundled by the build process in a single `[MODULE]-translations-bundle.js` file.
```

Comment:
Would it make sense to have it work for all "apps" and not just for a given module?
When updating translations we usually always want them all.